### PR TITLE
Distribution initial interfaces

### PR DESCRIPTION
**Pull request by a-holliday**
Ready for code review on CSVParser service, Email Service, and Auth Service tests, added documentation.

### CODE REVIEW

File: SurveyService/src/test/java/com/revature/servicetest/AuthServiceJWTTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	private final String SECRET = "secret";
+
+	/**
+	 * Checks AuthServiceJWT.createToken(surveyId) with a valid surveyId. If token
+	 * is generated with the correct claims for surveyId, batchId, IAT, and EXP,
+	 * test passes. For temporal claims (IAT, EXP), a reasonable time range is
+	 * checked in lieu of exact values.
+	 */
+	@Test
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/servicetest/AuthServiceJWTTest.java:40]: Spy on JWT parser, and verify appropriate methods were called to create token

File: SurveyService/src/test/java/com/revature/servicetest/AuthServiceJWTTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+
+import com.revature.service.AuthService;
+import com.revature.util.InvalidBatchIdException;
+import com.revature.util.InvalidSurveyIdException;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/servicetest/AuthServiceJWTTest.java:20]: Rewrite tests, make sure they are testing what is actually being validated in this service

File: SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import com.revature.service.CSVParser;
+
+@SpringBootTest
+class CSVParserTest {
+	@Autowired
+	CSVParser csvParser;
+
+	/**
+	 * Tests CSVParser service, creates a valid CSV and inputs it as parameter,
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java:36]: fix description, should be what this is testing, not what the test is doing

File: SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		MockMultipartFile emailFile = new MockMultipartFile("data", "nonexistingfile.csv", "text/plain",
+				"acacia.holliday@revature.net,ksenia.milstein@revature.net,zach.leonardo@revature.net".getBytes());
+		List<String> actualEmails = csvParser.parseFileForEmails(emailFile);
+		assertEquals(testEmails, actualEmails);
+
+	}
+
+	/**
+	 * Tests CSVParser service with invalid CSV file, should throw a
+	 * FileNotFoundException.
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java:52]: if CSV file is invalid but there, why is it throwing a FNF exception

File: SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import com.revature.service.CSVParser;
+
+@SpringBootTest
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/servicetest/CSVParserTest.java:30]: Test with more accurate data, we shouldn't have a batch with 3 associates

**Review by jurczakn — CHANGES_REQUESTED** [review]

**Review by jurczakn — APPROVED** [review]

**Comment by kmilstein** [comment]
Appears to be the same as branch feature-email-server

**Pull request closed without merging**